pkg/config/config.go

```diff
@@ -130,7 +213,41 @@ func ParseCliArgs() Config {
 		config.EnableSpotInterruptionDraining,
 	)
 
-	return config
+	return config, nil
+}
+
+func createFlags(flagData map[string]map[string]interface{}) (map[string]interface{}, error) {
+	result := make(map[string]interface{})
+
+	for name, data := range flagData {
+		switch data["defValue"].(type) {
+		case string:
+			value := getEnv(data["key"].(string), data["defValue"].(string))
```

Comment:
using a typed key field in a struct would also prevent a possible panic here if "key" for some reason was not a string.